site/content/extend/plugins/best-practices.md

```diff
@@ -6,6 +6,67 @@ weight: 50
 
 See here for [server-specific best practices for plugins](/extend/plugins/server/best-practices/). Webapp-specific best practices are incoming.
 
+## How can I create my own plugin setting?
+
+Plugins can define their admin settings in the plugin's manifest [setting schema](https://developers.mattermost.com/extend/plugins/manifest-reference/#settings_schema) which include a pre-defined types that the System Console UI currently supports. 
+
+Additionally, a plugin could define their own type of setting with a corresponding custom user interface that will be displayed in the system console following the process below. 
+
+1. Define a setting in the plugins manifest setting scheme with a type of `custom`
+
+```
+    "settings_schema": {
+        "settings": [{
+            "key": "CustomSetting",
+            "type": "custom"
+       }]
+```
+
+2. In the plugin Web App code, define a custom component to manage the plugin's specific setting and register in the Web App with [`registerAdminConsoleCustomSetting`](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerAdminConsoleCustomSetting). This component will be instantiated in the System Console with the following `props` passed in:
```

Comment:
```suggestion
2. In the plugin's Web App code, define a custom component to manage the plugin's custom setting and register in the Web App with [`registerAdminConsoleCustomSetting`](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerAdminConsoleCustomSetting). This component will be represented in the System Console with the following `props` passed in:
```

Reply:
Could we keep `instantiated` here? Given that I'm referring to specific `props` that are passed at the time the component is created not sure if `represented` would have the same meaning.